### ISSUE TITLE
Add "authorize" hook

### ISSUE THREAD

**Issue by ritch** [strongloop/loopback#1363]
In order to fix https://github.com/strongloop/loopback/issues/1191 we need support for a special hook that is executed as the first "before" hook. Something like this should work:

```js

remotes.authorize = function(ctx, next) {
   // called first, preceding any other before hook
   next();
}

```

`checkAccess()` would be called using this hook ensuring ACLs are checked before any other hooks.

/cc @fabien @bajtos @raymondfeng 

**Tasks**

 - [x] create [PR](https://github.com/strongloop/strong-remoting/pull/212) adding authorization hook to strong-remoting
 - [x] create [PR](https://github.com/strongloop/loopback/pull/1370) using the new authorization hook in `enableAuth()` instead of `beforeRemote`

**Comment by seriousben** [comment]
Really last, not as the first remote hook?

**Comment by ritch** [comment]
> Really last, not as the first remote hook?

edited :)

**Comment by seriousben** [comment]
:)

**Comment by bajtos** [comment]
As long as there is only one hook of this type, then it makes sense. Can we please use a different method name please, to make it more obvious what is the purpose of this hook? For example:

```js
remotes.authorize = function(ctx, next) { next(); };
```

**Comment by ritch** [comment]
`authorize` sounds good to me.

**Comment by ritch** [comment]
Landed this new feature. Released as **v2.17.3**
